### PR TITLE
Merge last dev with master

### PR DESCRIPTION
**Pull request by ImplicitDeny**
Once merged, this PR will :
- fix all spotted bugs on last session
- bring last opex template

### CODE REVIEW

File: functions/heli/fn_initHeli.sqf [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// No serverside task
```
Comment (ArwynFr) [inline functions/heli/fn_initHeli.sqf]: This file is too large IMHO

File: functions/heli/fn_initHeli.sqf [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+			// If names are the same, do not display
+			(([_newType] call GSRI_fnc_minifyName) != ([_heli] call GSRI_fnc_minifyName))
+		};
+		_action = [format["action%1", _display],_display,"",GSRI_fnc_spawnHeli,_condition,{},[_hangar, _x],"",2,[false, false, false, false, false], _modifier] call ace_interact_menu_fnc_createAction;
+		[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+	};
+} forEach _types;
+
+// Hangar cleaning action
```
Comment (ArwynFr) [inline functions/heli/fn_initHeli.sqf]: Comment denotes a good candidate for an external function

File: functions/heli/fn_initHeli.sqf [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	params ["_t", "_p", "_args"];
+	_args params ["_hangar"];
+	_heli = [_hangar] call GSRI_fnc_getHeli;
+	["HeliFRIES", [getText (configFile >> "CfgVehicles" >> typeOf _heli >> "displayName")]] call BIS_fnc_showNotification;
+	[_heli] remoteExecCall ["ace_fastroping_fnc_equipFRIES", 2];
+};
+_action = ["actionFRIES","Equiper FRIES","",_statement,_condition,{},[_hangar]] call ace_interact_menu_fnc_createAction;
+[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+// Place helicopter on rear deck
```
Comment (ArwynFr) [inline functions/heli/fn_initHeli.sqf]: Comment denotes a good candidate for an external function

File: functions/heli/fn_initHeli.sqf [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	params["_t", "_p", "_args"];
+	_args params ["_hangar", "_deck"];
+	_heli = [_hangar] call GSRI_fnc_getHeli;
+	["HeliMoved", [getText (configFile >> "CfgVehicles" >> typeOf _heli >> "displayName")]] call BIS_fnc_showNotification;
+	_heli setPosASL getPosASL _deck;
+};
+_action = ["actionDeck","Placer l'appareil sur le pont d'envol","",_statement,_condition,{},[_hangar, _deck]] call ace_interact_menu_fnc_createAction;
+[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+// Place helicopter in hangar
```
Comment (ArwynFr) [inline functions/heli/fn_initHeli.sqf]: Comment denotes a good candidate for an external function

File: functions/liberty/fn_libertyArsenal.sqf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+_start = [-2.42944,15.6,8.7];
+_actionArsenal = ["actionArsenal","Ouvrir le vestiaire","",{["Open",true] spawn bis_fnc_arsenal;},{true}] call ace_interact_menu_fnc_createAction;
+for [{_i = 0}, {_i < 9}, {_i = _i+1}] do {
+	_handle = "Land_Battery_F" createVehicleLocal [0,0,0];
+	_handle enableSimulation false;
+	_handle attachTo [_navire, (_start vectorAdd [0,(0.61*_i),0])];
+	[_handle, 0, [], _actionArsenal] call ace_interact_menu_fnc_addActionToObject;
+};
+
+// Second, add Preslots actions on grey lockers
```
Comment (ArwynFr) [inline functions/liberty/fn_libertyArsenal.sqf]: Comment denotes a good candidate for an external function

File: functions/liberty/fn_libertyBays.sqf [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	_tri = createVehicle ["Land_Destroyer_01_Boat_Rack_01_F", [0,0,0], [], 0, "NONE"];
+	_tri enableSimulation false;
+	_tri allowDamage false;
+	_tri setDir (getDir _navire + 180);
+	_tri setPosATL ASLToATL (_navire modelToWorldWorld [-11.5,14.43,7.5]);
+	_tri enableSimulation true;
+	[_tri] remoteExecCall ["BIS_fnc_BoatRack01Init", 0, true];
+	_navire setVariable ["baie_tribord", _tri, true];
+
+	// Baie babord
```
Comment (ArwynFr) [inline functions/liberty/fn_libertyBays.sqf]: Code duplication, denotes a good candidate for an external function

File: functions/liberty/fn_libertyInit.sqf [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+if(_sub) then { [_destroyer] call GSRI_fnc_initSubmarine };
+
+if(isServer) then {
+	// Add map marker
+	_mk = createMarker ["marker_destroyer", _destroyer];
+	_mk setMarkerType "flag_France";
+	_mk setMarkerText "D-651 Normandie";
+
+	// Add restricted zone sign
+	_sign = "SignAd_Sponsor_F" createVehicle [0,0,0];
```
Comment (ArwynFr) [inline functions/liberty/fn_libertyInit.sqf]: Code duplication denotes a good candidate for an external function

File: functions/submarine/fn_submarineSelectPos.sqf [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	params ["_units", "_pos", "_alt", "_shift"];
+
+	// Map click may happen but not linked to this module
+	if!(player getVariable ["submarine_flag", false]) exitWith {};
+
+	// Check if there is water here
+	_maxDepth = getTerrainHeightASL _pos;
+	if(_maxDepth > -25 and worldName != "VR") exitWith { openMap false; ["SubmarineFail"] call BIS_fnc_showNotification};
+
+	// TODO : check if there is players around the sub, both at actual and targeted coordinates
```
Comment (ArwynFr) [inline functions/submarine/fn_submarineSelectPos.sqf]: Create an issue for that ?

File: functions/zeus/fn_handleRequest.sqf [hunk 9]
```diff
@@ -11,5 +11,6 @@ switch (_array select 0) do {
 	case "slots" : { [] call GSRI_fnc_zeusSlots };
 	case "request" : { [] call GSRI_fnc_zeusRequest };
 	case "release" : { [] call GSRI_fnc_zeusRelease };
+	case "promote" : { [_array select 1] remoteExecCall ["GSRI_fnc_zeusPromote", 2] }; //TODO : ne gère pas les pseudo avec espace...
```
Comment (ArwynFr) [inline functions/zeus/fn_handleRequest.sqf]: Create an issue for that ?

**Comment by ImplicitDeny** [comment]
All reviews mentionning a language problem have to be solved at once : every kind of user-displayed text must be localized in accordance with the user's selected language. At least french and english translations can be provided.

**Comment by ArwynFr** [comment]
Hardcoded translations are ok for a quick release IMO, could be an issue that can be solved in september

**Comment by ImplicitDeny** [comment]
Changing the displayed sentence to english is not really less time-consuming that adding proper localization. Will do this quickly this week.

File: functions/submarine/fn_submarineSelectPos.sqf (hunk 8 above)
Comment (ImplicitDeny) [reply functions/submarine/fn_submarineSelectPos.sqf]: Done https://github.com/team-gsri/OpexTemplate/issues/9

File: functions/heli/fn_initHeli.sqf (hunk 3 above)
Comment (ImplicitDeny) [reply functions/heli/fn_initHeli.sqf]: this is a temporary function and will not be externalized

File: functions/heli/fn_initHeli.sqf (hunk 4 above)
Comment (ImplicitDeny) [reply functions/heli/fn_initHeli.sqf]: this is a temporary function and will not be externalized

File: functions/liberty/fn_libertyArsenal.sqf (hunk 5 above)
Comment (ImplicitDeny) [reply functions/liberty/fn_libertyArsenal.sqf]: Ehhh... Whut ?

Comment (ImplicitDeny) [reply functions/liberty/fn_libertyArsenal.sqf]: Fixed, see https://github.com/team-gsri/Orion/pull/10/commits/fd6773cb295ab7f77570d5654f5f847284cfd76f

**Pull request merged**